GenHyb/GenMethods.hs

```diff
@@ -0,0 +1,2729 @@
+{-# LANGUAGE MultiParamTypeClasses, FunctionalDependencies,
```

Comment:
this is a very large filel. Could it be split? E.g. parsing, static analysis, ...